.github/workflows/build-macos.yml

```diff
@@ -1,11 +1,15 @@
-on: 
+on:
   pull_request:
     branches:
       - master
   push:
     branches:
       - master
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Comment:
We don't have to keep these changes if you don't like the concept.

I feel like we get more CI failures for reasons that look like resource depletion (context timeouts as if the test had been paused for a while) when people make a multiple commits in succession to their pull request branch. CI will run tests to completion on every single commit causing many parallel jobs. The changes in this PR will cancel the previous job when the new commit comes in, and only run the tests on the latest commit of the current workflow. (Other workflows are unaffected ... 2 PRs will not cancel each other's tests.)

We have 2 workflows: (1) commits to pull request (2) commits to master. The second flow isn't likely to be affected by these changes, as it would be hard to have a 2nd pull request finish its tests in time and be allowed to merge to master before the previous master tests completed.

Downside 1: From a Github UI perspective, the cancelled tests from the previous commit will have a red `x` by them. ... just discovered that testing this PR. A minor nuisance, but better than the tests on the most recent commit failing due to resource depletion.

Downside 2: I don't see github documentation that resources are shared. I do feel that there is correlation, but I can't prove it without a doubt.

Reply:
seems ok, downside 1 is sort of annoying but can just wait for the job to finish before pushing i guess